# Đồ án cuối kỳ môn Lập trình cho Khoa học dữ liệu
## Giảng viên hướng dẫn: 
### Thầy Trần Trung Kiên
### Cô Phan Thị Phương Uyên
## Sinh viên thực hiện:
##### Dương Hoàng Quang Khanh, mã sinh viên: 1712521

---

## Import các thư viện cần thiết

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

## Về nguồn dữ liệu:
Dữ liệu được sử dụng trong bài tập này là dữ liệu về 20 nhóm mặt hàng xuất khẩu và 20 nhóm mặt hàng xuất khẩu chủ lực của Việt Nam với các quốc gia trên thế giới trong giai đoạn 2010-2019. Dữ liệu được thống kê bởi Phòng ban thống kê Liên Hợp Quốc (United Nations Statistics Division) và được em trích xuất và tải về từ trang https://comtrade.un.org/data. Dữ liệu được thống kê theo từng năm và được lưu trong file "comtrade_data.csv". 

## Khám phá dữ liệu
### Đọc dữ liệu từ file

In [14]:
pd.set_option('display.max_columns', None) #Giúp hiển thị tất cả các cột của dữ liệu
data = pd.read_csv("comtrade_data.csv")
data.head(5)

Classification  Year  Period  Period Desc.  Aggregate Level  \
0             H3  2010    2010          2010                2   
1             H3  2010    2010          2010                2   
2             H3  2010    2010          2010                2   
3             H3  2010    2010          2010                2   
4             H3  2010    2010          2010                2   

   Trade Flow Code Trade Flow  Reporter Code  Reporter Reporter ISO  \
0                0          M            704  Viet Nam          VNM   
1                0          M            704  Viet Nam          VNM   
2                0          M            704  Viet Nam          VNM   
3                0          M            704  Viet Nam          VNM   
4                0          M            704  Viet Nam          VNM   

   Partner Code    Partner Partner ISO  Commodity Code  \
0            32  Argentina         ARG               2   
1            36  Australia         AUS               2   
2            40    Austria         AUT               2   
3            76     Brazil         BRA               2   
4           124     Canada         CAN               2   

                    Commodity  Qty Unit Code  Qty Unit  Qty  \
0  Meat and edible meat offal             -1       NaN  NaN   
1  Meat and edible meat offal             -1       NaN  NaN   
2  Meat and edible meat offal             -1       NaN  NaN   
3  Meat and edible meat offal             -1       NaN  NaN   
4  Meat and edible meat offal             -1       NaN  NaN   

   Alt Qty Unit Code  Alt Qty Unit  Alt Qty  Netweight (kg)  \
0                 -1           NaN      NaN             NaN   
1                 -1           NaN      NaN             NaN   
2                 -1           NaN      NaN             NaN   
3                 -1           NaN      NaN             NaN   
4                 -1           NaN      NaN             NaN   

   Gross weight (kg)  Trade Value (US$)  CIF Trade Value (US$)  \
0                NaN            1705860              1705860.0   
1                NaN           10492280             10492280.0   
2                NaN              99310                99310.0   
3                NaN            5473690              5473690.0   
4                NaN             552850               552850.0   

   FOB Trade Value (US$)  Flag  
0                    NaN     0  
1                    NaN     0  
2                    NaN     0  
3                    NaN     0  
4                    NaN     0

### Kích thước của dữ liệu

In [9]:
print(data.shape)

(31520, 28)


Có thể thấy dữ liệu gồm có 31520 dòng và 28 cột.

Mỗi dòng dữ liệu mô tả về giá trị xuất khẩu hoặc nhập khẩu của Việt Nam với một đối tác về một nhóm mặt hàng trong một năm duy nhất.

Về mô tả thông tin của mỗi cột xin mời xem mô tả dưới đây:

- **Classification**: các phân loại quốc tế áp dụng cho dữ liệu thương mại và thuế quan được đưa ra bởi tổ chức Hải quan thế giới (WCO), H3 áp dụng cho hàng hóa từ 2007 - 2012, H4 là từ 2012 - 2017, và H5 là từ 2017 đến nay.
- **Year, Period, Period Desc**: Vì dữ liệu thông kê theo từng năm nên cả ba cột này đều mô tả năm mà dữu liệu được thống kê.
- **Aggregate Level**: đề cập đến mức độ chi tiết trong thống kê dữ liệu.
- **Trade Flow Code, Trade Flow**: 

In [13]:
data["Is Leaf Code"].unique()

array([0], dtype=int64)